src/khoj/routers/helpers.py

```diff
@@ -1186,3 +1204,286 @@ def construct_automation_created_message(automation: Job, crontime: str, query_t
     """.strip()
+
+
+def get_user_config(user: KhojUser, request: Request, is_detailed: bool = False):
+    user_picture = request.session.get("user", {}).get("picture")
+    is_active = has_required_scope(request, ["premium"])
+    has_documents = EntryAdapters.user_has_entries(user=user)
+
+    if not is_detailed:
+        return {
+            "request": request,
+            "username": user.username if user else None,
+            "user_photo": user_picture,
+            "is_active": is_active,
+            "has_documents": has_documents,
+            "khoj_version": state.khoj_version,
+        }
+
+    user_subscription_state = get_user_subscription_state(user.email)
+    user_subscription = adapters.get_user_subscription(user.email)
+    subscription_renewal_date = (
+        user_subscription.renewal_date.strftime("%d %b %Y")
+        if user_subscription and user_subscription.renewal_date
+        else (user_subscription.created_at + timedelta(days=7)).strftime("%d %b %Y")
+    )
+    given_name = get_user_name(user)
+
+    enabled_content_source = set(EntryAdapters.get_unique_file_sources(user))
+    successfully_configured = {
+        "computer": ("computer" in enabled_content_source),
+        "github": ("github" in enabled_content_source),
+        "notion": ("notion" in enabled_content_source),
+    }
+
+    selected_conversation_config = ConversationAdapters.get_conversation_config(user)
+    conversation_options = ConversationAdapters.get_conversation_processor_options().all()
+    all_conversation_options = list()
+    for conversation_option in conversation_options:
+        all_conversation_options.append({"chat_model": conversation_option.chat_model, "id": conversation_option.id})
+
+    search_model_options = adapters.get_or_create_search_models().all()
+    all_search_model_options = list()
+    for search_model_option in search_model_options:
+        all_search_model_options.append({"name": search_model_option.name, "id": search_model_option.id})
+
+    current_search_model_option = adapters.get_user_search_model_or_default(user)
+
+    selected_paint_model_config = ConversationAdapters.get_user_text_to_image_model_config(user)
+    paint_model_options = ConversationAdapters.get_text_to_image_model_options().all()
+    all_paint_model_options = list()
+    for paint_model in paint_model_options:
+        all_paint_model_options.append({"model_name": paint_model.model_name, "id": paint_model.id})
+
+    notion_oauth_url = get_notion_auth_url(user)
+
+    eleven_labs_enabled = is_eleven_labs_enabled()
+
+    voice_models = ConversationAdapters.get_voice_model_options()
+    voice_model_options = list()
+    for voice_model in voice_models:
+        voice_model_options.append({"name": voice_model.name, "id": voice_model.model_id})
+
+    if len(voice_model_options) == 0:
+        eleven_labs_enabled = False
+
+    selected_voice_config = ConversationAdapters.get_voice_model_config(user)
+
+    return {
```

Comment:
Oh man, lol.

Reply:
Yeah, there are a lot of them! I've arranged them with comment headers to make them a bit more readable